### PR TITLE
Isolate exceptions during search type result extraction.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This PR makes sure that if an exception is being thrown during extracting the results for a search type, it will be caught, added as as earch type error to the search result and extraction for other search types will be continued (similar to what we do already during search _generation_).

This prevents the error being handled as a fatal one (affecting the complete search), so other results will still be returned.

## Screenshots (if appropriate):

Before:
image

After:
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by luk-kaminski — APPROVED** [review]
LGTM, thank you!

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/views/ElasticsearchBackend.java [hunk 1]
```diff
@@ -269,9 +270,14 @@ public QueryResult doRun(SearchJob job, Query query, ESGeneratedQueryContext que
...
-                final SearchType.Result searchTypeResult = handler.extractResult(job, query, searchType, multiSearchResponse.getResponse(), queryContext);
-                if (searchTypeResult != null) {
-                    resultsMap.put(searchTypeId, searchTypeResult);
+                try {
+                    final SearchType.Result searchTypeResult = handler.extractResult(job, query, searchType, multiSearchResponse.getResponse(), queryContext);
+                    if (searchTypeResult != null) {
+                        resultsMap.put(searchTypeId, searchTypeResult);
+                    }
+                } catch (Exception e) {
+                    LOG.debug("Unable to extract results: ", e);
```
Comment (janheise) [inline graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/views/ElasticsearchBackend.java:279]: could be a warning?

File: graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/views/OpenSearchBackend.java [hunk 2]
```diff
@@ -269,10 +269,16 @@ public QueryResult doRun(SearchJob job, Query query, OSGeneratedQueryContext que
...
-                final SearchType.Result searchTypeResult = handler.extractResult(job, query, searchType, multiSearchResponse.getResponse(), queryContext);
-                if (searchTypeResult != null) {
-                    resultsMap.put(searchTypeId, searchTypeResult);
+                try {
+                    final SearchType.Result searchTypeResult = handler.extractResult(job, query, searchType, multiSearchResponse.getResponse(), queryContext);
+                    if (searchTypeResult != null) {
+                        resultsMap.put(searchTypeId, searchTypeResult);
+                    }
+                } catch (Exception e) {
+                    LOG.debug("Unable to extract results: ", e);
```
Comment (janheise) [inline graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/views/OpenSearchBackend.java:278]: could be a warning?

**Review by janheise — APPROVED** [review]
LGTM!

**Pull request merged**